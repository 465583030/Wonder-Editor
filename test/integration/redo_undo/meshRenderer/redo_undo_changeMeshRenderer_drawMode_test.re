open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: change meshRenderer drawMode", () => {
    let sandbox = getSandboxDefaultVal();

    let _getFromArray = (array, index) => ArrayService.unsafeGetNth(index, array);

    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initState(~sandbox, ());

      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
    let _simulateChangeTwiceDrawMode = () => {
      let lineType = MainEditorMeshRendererTool.getDrawModeLineType();

      MainEditorMeshRendererTool.triggerChangeDrawModeEvent(lineType);

      let triangleFanType =
        MainEditorMeshRendererTool.getDrawModeTriangleFanType();

      MainEditorMeshRendererTool.triggerChangeDrawModeEvent(triangleFanType);
    };

    let _beforeEach = () =>
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        () => {
          MainEditorAssetTool.initAssetTree();
          MainEditorSceneTool.setFirstBoxTobeCurrentSceneTreeNode();
        },
      );

    RedoUndoTool.testRedoUndoTwoStep(
      sandbox,
      "prepare first step: set currentSceneTreeNode",
      (_simulateChangeTwiceDrawMode, _beforeEach, () => ()),
      BuildComponentForCurryTool.buildMeshRenderer,
    );
  });