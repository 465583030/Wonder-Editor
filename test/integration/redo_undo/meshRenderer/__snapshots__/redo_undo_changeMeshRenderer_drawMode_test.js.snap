// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: change meshRenderer drawMode prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="1"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`redo_undo: change meshRenderer drawMode prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="6"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`redo_undo: change meshRenderer drawMode prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="6"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`redo_undo: change meshRenderer drawMode prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="1"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`redo_undo: change meshRenderer drawMode prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="4"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;
